### ISSUE TITLE
Popup stdout

### ISSUE THREAD

**Issue by liquidz** [liquidz/vim-iced#300]
Branched from https://github.com/liquidz/vim-iced/issues/291#issuecomment-735310376

**Comment by tami5** [comment]
:heart_eyes: :heart_eyes:  :heart_eyes:

**Comment by liquidz** [comment]
prototype
![output](https://user-images.githubusercontent.com/6941/102667072-de610980-41cb-11eb-8b29-2fa245c6f51e.gif)

**Comment by tami5** [comment]
Oh wow very minimal I like it. Ideas:

- Syntax highlighting
- print hello world as a string like `"Hello world"`
- keep popup until user move cursor 
- wrap long lines
- have one size (in terms of width)
- have a title like : `;; eval (println "hello\nworld)"` 

Well done @liquidz great progress

**Comment by liquidz** [comment]
@tami5 
> keep popup until user move cursor

I think it isn't suitable for async output.

```clojure
(ns foo.core
  (:require
   [clojure.core.async :as async]))

;; If these code are wrapped as a function,
;; user cannot know when the "foo" will be outputted.
(async/go
  (async/<! (async/timeout 3000))
  (println "foo"))
```

> have a title like : ;; eval (println "hello\nworld)"

As a same reason, it isn't suitable for async output.
It's not impossible, but the process will be complicated.

**Comment by tami5** [comment]
@liquidz whats more important is that stdout buffer becomes interactive recording not println function but also the output of evaluating forms in general. As it can be a good place to test/evaluate things out without effects the main buffer.

**Comment by liquidz** [comment]
@tami5 Sorry for late reply.

At first, we should consider the roles of "stdout buffer" and "popup stdout"(temporary) separately.
For me, the role of "popup stdout" is to notice that standard output was there.
In other words, a trigger to refer to "stdout buffer" as needed.

A small amount of output can be checked by "popup stdout".
However, the display area of "popup stdout" is limited, so "stdout buffer" should take the role of checking the entire output.

Because of the above, I'd like to minimize the display of "popup stdout".
So, I have no plan to adopt the following for now.

> wrap long lines
> have a title like : ;; eval (println "hello\nworld)"

**Comment by tami5** [comment]
Oh sorry for mixing the two together. 

> A small amount of output can be checked by "popup stdout".
> However, the display area of "popup stdout" is limited, so "stdout buffer" should take the role of checking the entire output.
> 
> Because of the above, I'd like to minimize the display of "popup stdout".
> So, I have no plan to adopt the following for now.
> 
> > wrap long lines
> > have a title like : ;; eval (println "hello\nworld)"

Very good point, we need to direct our attention to the role of "popup stdout". 

I think I begin to see what you are making the popup stdout role to be, and I belive it is to go hand in hand with the virtual text.

Looking forward for improvement on "stdout buffer" :D, it just sometime long maps with the virtual text isn't sufficient . Thanks @liquidz

**Comment by liquidz** [comment]
Closed by #304
